Source/swiftlint/Commands/LintCommand.swift

```diff
@@ -119,3 +122,27 @@ struct LintOptions: OptionsType {
             <*> mode <| quietOption(action: "linting")
     }
 }
+
+private func isWarningThresholdBroken(configuration: Configuration,
+                                      violations: [StyleViolation]) -> Bool {
+    guard let warningThreshold = configuration.warningThreshold else { return false }
+    let numberOfWarningViolations = violations.filter({ $0.severity == .Warning}).count
+    if numberOfWarningViolations >= warningThreshold {
```

Comment:
these 4 lines should just be `return numberOfWarningViolations >= warningThreshold`.